packages/opentelemetry-core/src/trace/TracerDelegate.ts

```diff
@@ -23,57 +23,45 @@ import { NoopTracer } from './NoopTracer';
 // applications with a complex set of dependencies. Also allows for the tracer
 // to be changed/disabled during runtime without needing to change reference
 // to the global tracer
-export class GlobalTracerDelegate implements types.Tracer {
-  private _tracer: types.Tracer | null;
-  private readonly _fallbackTracer: types.Tracer;
+export class TracerDelegate implements types.Tracer {
+  private readonly _tracer: types.Tracer;
 
-  // Wrap a tracer with a GlobalTracerDelegate. Provided tracer becomes the default
+  // Wrap a tracer with a TracerDelegate. Provided tracer becomes the default
   // fallback tracer for when a global tracer has not been initialized
-  constructor(fallbackTracer?: types.Tracer) {
-    this._tracer = null;
-    this._fallbackTracer = fallbackTracer || new NoopTracer();
-  }
-
-  set tracer(tracer: types.Tracer | null) {
-    this._tracer = tracer;
+  constructor(tracer?: types.Tracer | null, fallbackTracer?: types.Tracer) {
```

Comment:
What if we did a default argument of `fallbackTracer: types.Tracer = new NoopTracer()`? And then in the body just have `this._tracer = tracer || fallbackTracer`?

Reply:
Refactored to use typescript constructor class field initializers. I want the user specified Tracer arg to be called `tracer` and fallback is called `fallbackTracer`. Therefore, the current tracer the delegate is using (via `start`/`stop`) is now called `_currentTracer`